src/dialectic/core.py

```
@@ -0,0 +1,305 @@
+
+
+class DialecticAgent:
+    """
+    An agentic dialectic that iteratively gathers context to answer queries.
+
+    Unlike the standard dialectic which pre-gathers all context before a single
+    LLM call, this agent uses tools to strategically gather only the context
+    needed to answer the specific query.
+    """
+
+    def __init__(
```

Ruff (0.14.10)

Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)